sdk/lib/opentelemetry/sdk/trace/samplers/probability_sampler.rb

```diff
@@ -8,8 +8,88 @@ module OpenTelemetry
   module SDK
     module Trace
       module Samplers
-        class ProbabilitySampler
-          # TODO
+        # A {OpenTelemetry::Trace::Samplers::Sampler} where the probability of
+        # sampling a trace is equal to that of the specified probability.
+        class ProbabilitySampler < OpenTelemetry::Trace::Samplers::Sampler
+          class << self
+            private :new # rubocop:disable Style/AccessModifierDeclarations
+
+            # Returns a new {OpenTelemetry::Trace::Samplers::Sampler}. The
+            # probability of sampling a trace is equal to that of the specified
+            # probability.
+            #
+            # @param [Numeric] probability The desired probability of sampling.
+            #   Must be within [0.0, 1.0].
+            # @raise [ArgumentError] if probability is out of range
+            # @return [OpenTelemetry::Trace::Samplers::Sampler]
+            def create(probability)
+              raise ArgumentError, 'probability must be in range [0.0, 1.0]' unless (0.0..1.0).include?(probability)
+
+              return OpenTelemetry::Trace::Samplers::NeverSampleSampler.new if probability == 0.0
```

Comment:
This interferes with the logic in `decision`. Specifically, it does not respect sampling decisions made by the parent span or links. Which raises a couple of interesting questions:
1. Should `probability == 0` be equivalent to "never sample", even in the face of upstream decisions?
2. Should `NeverSampleSampler` respect upstream decisions?

The spec currently says nothing about `ProbabilitySampler`. It *is* specific about [`NeverSampleSampler`](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-sampling.md):

> * Never sample. `Sampler` returns `Decision` with `IsSampled=false` and empty arguments collection. Description MUST be `NeverSampleSampler`.

Reply:
These are good questions. I'll see if I can get some clarification at the spec meeting tomorrow.

Reply:
The [sampling OTEP](https://github.com/open-telemetry/oteps/blob/master/text/0006-sampling.md#default-samplers) has some additional information. From my read, the NeverSampleSampler (ALWAYS_OFF) sampler should not respect upstream decisions. It also says the probability sampler can be configured to ignore the parent sampling decision. So, it would be equivalent to the NeverSampleSampler when probability is 0.0 and its configured to ignore the parent sampling decision.

It still doesn't tell us exactly what we should do for the "classic" sampling specification, which is what we're currently working with. Since we don't have the ability to configure whether or not to ignore the parent sampling decision, we should probably respect it. Alternatively, we could probably just go with what we have if it's going to be replaced by the OTEP anyway.

Reply:
I removed it. We can figure out the right thing to do when we implement the OTEP.